Makefile

```diff
@@ -60,11 +60,27 @@ build: GOFLAGS += -i -o cockroach
 build: BUILDMODE = build
 build: install
 
+.PHONY: checkgopath
+checkgopath:
+	@GOPATH_ABS=$$(readlink -f "$$GOPATH"); \
+	if [ ! -d "$$GOPATH_ABS" ]; then \
+	   echo 'Invalid GOPATH. Please make sure GOPATH is set correctly.'; \
+	   exit 1; \
+	fi; \
+	MAKEFILE_ABS=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST)))); \
+	if [ "$${MAKEFILE_ABS#$$GOPATH_ABS}" = "$$MAKEFILE_ABS" ]; then \
+	   echo "Makefile directory not under GOPATH!"; \
+	   echo "GOPATH: $$GOPATH"; \
+	   echo "Makefile directory: $$MAKEFILE_ABS"; \
+	   echo "Please make sure GOPATH is set correctly."; \
+	   exit 1; \
+	fi
+
 .PHONY: install
 install: LDFLAGS += -X "github.com/cockroachdb/cockroach/util.buildTag=$(shell git describe --dirty --tags)"
 install: LDFLAGS += -X "github.com/cockroachdb/cockroach/util.buildTime=$(shell date -u '+%Y/%m/%d %H:%M:%S')"
 install: LDFLAGS += -X "github.com/cockroachdb/cockroach/util.buildDeps=$(shell GOPATH=${GOPATH} build/depvers.sh)"
-install:
+install: checkgopath
```

Comment:
Look at the way we handle `.bootstrap` for a trick to do something for every `make` run without adding a dependency to every target.